pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go

```diff
@@ -411,7 +420,8 @@ func (s *Scheduler) Decide(ctx context.Context, labels labels.Labels) iface.Limi
 
 	reqCtx := ctx
 
-	if clientDeadline, hasDeadline := ctx.Deadline(); hasDeadline {
+	clientDeadline, hasDeadline := ctx.Deadline()
+	if hasDeadline || hasWorkloadTimeout {
```

Comment:
`clientDeadline` is valid only when `hasDeadline == true`. Cannot make the calculation at https://github.com/fluxninja/aperture/pull/2326/files#diff-9bad7db7432925c3ff46ac145c61f11d0733bc81463f2c6c85c082f95b4f8e95R431

Reply:
Addressed in https://github.com/fluxninja/aperture/pull/2336.